deepchem/models/torch_models/text_cnn.py

```diff
@@ -179,3 +186,210 @@ def forward(self, input: OneOrMany[torch.Tensor]) -> List[Any]:
+                                               pad_batches=pad_batches):
+                if y_b is not None:
+                    if self.mode == 'classification':
+                        y_b = to_one_hot(y_b.flatten(),
+                                         2).reshape(-1, self.n_tasks, 2)
+                # Transform SMILES sequence to integers
+
+                X_b = self.smiles_to_seq_batch(ids_b)
+                yield ([X_b], [y_b], [w_b])
+
+    @staticmethod
+    def build_char_dict(dataset: Dataset,
+                        default_dict: Dict[str, int] = default_dict):
+        """ Collect all unique characters(in smiles) from the dataset.
+       This method should be called before defining the model to build appropriate char_dict
+       """
+        X = dataset.ids
+        # Maximum length is expanded to allow length variation during train and inference
+        seq_length = int(max([len(smile) for smile in X]) * 1.2)
+        # '_' served as delimiter and padding
+        all_smiles = '_'.join(X)
+        tot_len = len(all_smiles)
+        # Initialize common characters as keys
+        keys = list(default_dict.keys())
+        out_dict = copy.deepcopy(default_dict)
+        current_key_val = len(keys) + 1
+        # Include space to avoid extra keys
+        keys.extend([' '])
+        extra_keys = []
+        i = 0
+        while i < tot_len:
+            # For 'Cl', 'Br', etc.
+            if all_smiles[i:i + 2] in keys:
+                i = i + 2
+            elif all_smiles[i:i + 1] in keys:
+                i = i + 1
+            else:
+                # Character not recognized, add to extra_keys
+                extra_keys.append(all_smiles[i])
+                keys.append(all_smiles[i])
+                i = i + 1
+        # Add all extra_keys to char_dict
+        for extra_key in extra_keys:
+            out_dict[extra_key] = current_key_val
+            current_key_val += 1
+        return out_dict, seq_length
+
+    def smiles_to_seq(self, smiles: str):
+        """
+        Tokenize characters in smiles to integers
+        """
+        smiles_len = len(smiles)
+        seq = [0]
+        keys = self.char_dict.keys()
+        i = 0
+        while i < smiles_len:
+            # Skip all spaces
+            if smiles[i:i + 1] == ' ':
+                i = i + 1
+            # For 'Cl', 'Br', etc.
+            elif smiles[i:i + 2] in keys:
+                seq.append(self.char_dict[smiles[i:i + 2]])
+                i = i + 2
+            elif smiles[i:i + 1] in keys:
+                seq.append(self.char_dict[smiles[i:i + 1]])
+                i = i + 1
+            else:
+                raise ValueError('character not found in dict')
+        for i in range(self.seq_length - len(seq)):
+            # Padding with '_'
+            seq.append(self.char_dict['_'])
+        return np.array(seq, dtype=np.int32)
+
+    @staticmethod
+    def convert_bytes_to_char(s):
```

Comment:
Documentation on parameters to be added
type annotations missing

Reply:
Added parameter descriptions and type annotations.